Set the SONAME for the libraries. Configure not to set RPATH to any
built lib/bin object. Force building share object libraries.--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,6 +38,11 @@
 
 
 cmake_minimum_required(VERSION 2.6.3 FATAL_ERROR)
+
+# No RPATH at all in any of the binaries built here:
+set (CMAKE_SKIP_RPATH TRUE)
+MESSAGE(STATUS "CMAKE_SKIP_RPATH: ${CMAKE_SKIP_RPATH}")
+
 SET(CMAKE_ALLOW_LOOSE_LOOP_CONSTRUCTS true)
 set(CF_OPENMS_PACKAGE_VERSION "${OPENMS_PACKAGE_VERSION_MAJOR}.${OPENMS_PACKAGE_VERSION_MINOR}.${OPENMS_PACKAGE_VERSION_PATCH}" CACHE INTERNAL "OpenMS VERSION" FORCE)
 
@@ -337,15 +342,20 @@
 if (FOUND_CUDART AND MT_ENABLE_CUDA) # we need both conditions due to possible present cached entries
 	CUDA_ADD_LIBRARY(OpenMS ${OpenMS_sources} ${Cuda_sources})
 else()
-	ADD_LIBRARY(OpenMS ${OpenMS_sources})
+	ADD_LIBRARY(OpenMS SHARED ${OpenMS_sources})
 endif()
 
+set_target_properties(OpenMS PROPERTIES VERSION 1.9.0 SOVERSION 1)
+
 ### OpenMS_GUI lib
-ADD_LIBRARY(OpenMS_GUI ${OpenMSVisual_sources})
+ADD_LIBRARY(OpenMS_GUI SHARED ${OpenMSVisual_sources})
 if (MSVC)
 	set (GUI_lnk_flags "/FORCE:MULTIPLE /INCREMENTAL:NO /ignore:4006 /ignore:4088")
 	set_target_properties(OpenMS_GUI PROPERTIES LINK_FLAGS ${GUI_lnk_flags}) ## allow multiple definitions of symbols (e.g. from template instanciations or STL-derived classes)
-endif()	
+endif()
+
+set_target_properties(OpenMS_GUI PROPERTIES VERSION 1.9.0 SOVERSION 1)
+	
 
 ## update the openms_svn_revision.h file before compiling OpenMS
 add_custom_target(svn_revision_update ALL
